sdk/identity/azure-identity/test/ut/simplified_header_test.cpp

```diff
@@ -9,17 +9,46 @@
  */
 
 #include <azure/identity.hpp>
-#include <gtest/gtest.h>
 
-class DllExportTest final {
-  AZ_IDENTITY_DLLEXPORT static const bool DllExportHIncluded;
-};
+#include <gtest/gtest.h>
 
 TEST(SimplifiedHeader, identity)
 {
   using namespace Azure::Identity;
 
-  EXPECT_NO_THROW(ClientSecretCredential clientSecretCredential("", "", ""));
-  EXPECT_NO_THROW(EnvironmentCredential environmentCredential);
-  EXPECT_NO_THROW(static_cast<void>(static_cast<ManagedIdentityCredential const*>(nullptr)));
+  static_assert(
+      sizeof(AzureCliCredential) != 0,
+      "azure/identity.hpp does not include azure_cli_credential.hpp");
+
+  static_assert(
+      sizeof(ChainedTokenCredential) != 0,
+      "azure/identity.hpp does not include chained_token_credential.hpp");
+
+  static_assert(
+      sizeof(ClientCertificateCredential) != 0,
+      "azure/identity.hpp does not include client_certificate_credential.hpp");
+
+  static_assert(
+      sizeof(ClientSecretCredential) != 0,
```

Comment:
Can't this be achieved by simply declaring a local variable of type ClientSecretCredential?

I'm pretty sure that the static_assert won't compile if ClientSecretCredential is defined but not declared, so adding the static_assert doesn't add much value IMHO (because if CSC isn't defined, the code won't compile in the first place).

Reply:
Some constructors need parameters, so sizeof() is more minimalistic way to check for the type for being defined.

Reply:
That's fair.

Reply:
You're right that compilation will fail before it even gets to display the static_seert message - I'll rewrite that to check via `static_cast<void>(sizeof(X))`.